packages/explat-client/src/internal/experiment-assignment-store.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Internal dependencies
+ */
+import type { ExperimentAssignment } from '../types';
+import * as Validations from './validations';
+
+/**
+ * Class to hold ExperimentAssignment state
+ */
+export default class ExperimentAssignmentStore {
+	experimentAssigntmentsByExperimentName: Record< string, ExperimentAssignment | undefined > = {};
```

Comment:
Shall we make this `private`?

Also, `Assigntments` has a typo, but it may be better to go for a shorter name for the mapping like `experimentNameToAssignment`.